src/main/java/teammates/ui/webapi/ResetAccountRequestAction.java

```diff
@@ -19,36 +23,41 @@ class ResetAccountRequestAction extends AdminOnlyAction {
 
     @Override
     public JsonResult execute() throws InvalidOperationException {
-        String instructorEmail = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_EMAIL);
+        String email = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_EMAIL);
         String institute = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_INSTITUTION);
 
-        AccountRequestAttributes accountRequest = logic.getAccountRequest(instructorEmail, institute);
+        AccountRequestAttributes accountRequest = logic.getAccountRequest(email, institute);
 
         if (accountRequest == null) {
-            throw new EntityNotFoundException("Account request for instructor with email: " + instructorEmail
+            throw new EntityNotFoundException("Account request with email: " + email
                     + " and institute: " + institute + " does not exist.");
         }
 
-        if (accountRequest.getRegisteredAt() == null) {
+        if (!accountRequest.hasRegistrationKeyBeenUsedToJoin()) {
             throw new InvalidOperationException("Unable to reset account request as instructor is still unregistered.");
         }
 
         try {
             accountRequest = logic.updateAccountRequest(AccountRequestAttributes
-                .updateOptionsBuilder(instructorEmail, institute)
+                .updateOptionsBuilder(email, institute)
+                .withStatus(AccountRequestStatus.APPROVED)
+                .withLastProcessedAt(Instant.now())
                 .withRegisteredAt(null)
                 .build());
         } catch (InvalidParametersException | EntityDoesNotExistException e) {
             // InvalidParametersException should not be thrown as validity of params verified when fetching entity.
             // EntityDoesNoExistException shuold not be thrown as existence of entity has just been validated.
             log.severe("Unexpected error", e);
             return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (EntityAlreadyExistsException e) {
+            log.severe("The account request is not supposed to be re-created.", e);
```

Comment:
combine with previous block.

Reply:
I'm fine with this, especially because this action is going to be deleted.